### PR TITLE
Cpo 1740 swt new jersey

### PR DESCRIPTION
**Pull request by cstait**

### CODE REVIEW

File: CertiPay.Taxes.State/New Jersey/TaxTable.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          
+            taxableWages -= GetPersonalAllowance(personalAllowances);            
+
+            var selected_row = GetTaxWithholding(filingStatus, taxableWages);
+
+            // Calculate the withholding from the percentages
+
+            var taxWithheld = selected_row.TaxBase + ((taxableWages - selected_row.StartingAmount) * selected_row.TaxRate);
+
+            //(5) If zero exemption is claimed, subtract the standard deduction only.
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/New Jersey/TaxTable.cs]: What's this referring to?

File: CertiPay.Taxes.State/New Jersey/TaxTable.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            return
+                TaxableWithholdings
+                .Where(d => d.FilingStatus == filingStatus)
+                .Where(d => d.StartingAmount <= taxableWages)
+                .Where(d => taxableWages < d.MaximumWage)
+                .Select(d => d)
+                .Single();
+        }
+
+        public class StandardDeduction : DependentAllowance
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/New Jersey/TaxTable.cs]: Is this used?

File: .vs/restore.dg [hunk 3]
```diff
@@ -1,3 +1,3 @@
-#:C:\Users\Administrator\Documents\GitHub\CertiPay.Taxes.State\CertiPay.Taxes.State\CertiPay.Taxes.State.xproj
-#:C:\Users\Administrator\Documents\GitHub\CertiPay.Taxes.State\CertiPay.Taxes.State.Tests\CertiPay.Taxes.State.Tests.xproj
-C:\Users\Administrator\Documents\GitHub\CertiPay.Taxes.State\CertiPay.Taxes.State.Tests\CertiPay.Taxes.State.Tests.xproj|C:\Users\Administrator\Documents\GitHub\CertiPay.Taxes.State\CertiPay.Taxes.State\CertiPay.Taxes.State.xproj
+#:C:\Users\ctait\Documents\CertiPay.Taxes.State\CertiPay.Taxes.State.Tests\CertiPay.Taxes.State.Tests.xproj
```
Comment (mattgwagner) [inline .vs/restore.dg]: Whoops, I probably didn't mean to commit this. I'll remove it later.

File: CertiPay.Taxes.State/New Jersey/TaxTable.cs (hunk 1 above)
Comment (cstait) [inline CertiPay.Taxes.State/New Jersey/TaxTable.cs]: lazy copy paste from GA

File: CertiPay.Taxes.State/New Jersey/TaxTable.cs (hunk 2 above)
Comment (cstait) [inline CertiPay.Taxes.State/New Jersey/TaxTable.cs]: no, no dependant allowances

File: CertiPay.Taxes.State/New Jersey/TaxTable.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace CertiPay.Taxes.State.NewJersey
+{
+    public abstract class TaxTable : TaxTableHeader
+    {
+        public override StateOrProvince State { get { return StateOrProvince.NJ; } }
+        
+        public abstract IEnumerable<StandardDeduction> PersonalAllowances { get; }
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/New Jersey/TaxTable.cs]: You hardcoded `1000` for personal allowances below, making this kind of irrelevant?

**Pull request merged**